plugins/modules/cargo.py

```diff
@@ -43,6 +43,12 @@
       The base path where to install the Rust packages. Cargo automatically appends
       V(/bin). In other words, V(/usr/local) will become V(/usr/local/bin).
     type: path
+  registry:
+    description:
+      - Specify which registry to use.
```

Comment:
1. Is this only relevant during install? I.e. does rust remember where it installed a package from?
2. What happens if a user wants to change the registry for an already installed package? The code change does not seem to consider that case (I have no idea why it is actually possible). I guess it would be good to add another paragraph to explicitly state that that won't happen:
```suggestion
      - Specify which registry to use.
      - This is used only when installing new packages or installing another version of a package.
```
Or would it make sense to also query the repository (next to the version) where a package was installed from, and reinstall on mismatch?

Reply:
1. It does remember where the installed packages is from, but you always need to specify registry, or cargo will use the default one. So if you use private registry to install a package, but then update it without the argument, cargo would revert it to the official registry.
2. Cargo will reinstall it from a different registry. I think it would makes sense to actually check the registry and reinstall on mismatch as you've suggested.

I'll make another commit to address these.

Reply:
There's a caveat with `cargo` output when using custom registry. When you run `cargo install --registry ...`, you need to refer to the registry by its name. However, when showing installed packages with `cargo install --list`, `cargo` show the registry URL. The registry configuration is stored in `config.toml` and unfortunately it's not a simple mapping. So you would need TOML parser to figure out which registry name corresponds with which URL. But `tomllib` has been in Python only since 3.11. So it looks like the best course of action would be take registry into account only when installing new package/version as per your commit suggestion. Let me know your thoughts.

Reply:
You can also use tomli or toml from PyPI. They work relatively similar, see https://github.com/ansible-community/antsibull-changelog/blob/main/src/antsibull_changelog/toml.py for example (that code is for Python 3.9+, the `encoding` parameter for `open` doesn't work with Python 2.7 for example, so when supporting more Python versions it needs to be done slightly different).

How about requiring one of tomllib, tomli, and toml when `registry` is specified? Only when it is specified the registry needs to be considered for idempotency checks anyway.

Reply:
Where would be the proper place for the TOML code in the collection? `cargo.py` does not seem like the right place.

Reply:
How about `plugins/module_utils/toml.py`?